src/Common/src/Interop/Unix/System.Native/Interop.Read.cs

```diff
@@ -18,6 +19,6 @@ internal static partial class Sys
         /// Note - on fail. the position of the stream may change depending on the platform; consult man 2 read for more info
         /// </returns>
         [DllImport(Libraries.SystemNative, SetLastError = true)]
-        internal static unsafe extern int Read(int fd, byte* buffer, int count);
+        internal static unsafe extern int Read(SafeFileHandle fd, byte* buffer, int count);
```

Comment:
My thinking/reasoning was type-safety/discoverability.  I know this is internal-only code, but it felt odd that Read could take *any* SafeHandle, i.e. not a File or Pipe handle.

I can change it, if you'd like, but that is where my head was at.